### PR TITLE
Sort Markets by Title 

### LINKED ISSUE #285: Use a stable sort for entries in cities.json for easier diffs
> ... always sort the items in some reproducible manner so that the diffs are a bit shorter and easier to read? Now they sometimes seem bigger than they actually are because the order of items changed.

See https://github.com/wo-ist-markt/wo-ist-markt.github.io/pull/283#discussion_r258354322

### PR DESCRIPTION
**Pull request by NWuensche**
Resolves #285 .

### CODE REVIEW

**Comment by johnjohndoe** [comment]
I would prefer if the formatting would be applied to the file which we download from the data source, e.g. [`markets-berlin.json`](https://github.com/wo-ist-markt/wo-ist-markt.github.io/blob/master/preprocessing/berlin/raw/markets-berlin.json). The `berlin.json` file which is generated should then be updated accordingly.

**Comment by johnjohndoe** [comment]
@NWuensche Do you want to continue with your work or should we close this pull request?

**Comment by NWuensche** [comment]
Sorry, I have completely forgotten about this. I will try to fix this in the next days.

**Comment by NWuensche** [comment]
Should be fixed now. I sort and rewrite the `markets-berlin.json` file now.

File: cities/berlin.json [hunk 1]
```diff
@@ -2147,249 +2163,243 @@
...
             },
             "type": "Feature"
         }
     ],
-    "metadata": {
-        "data_source": {
-            "title": "Stadt Berlin, CC BY 3.0 DE, aktualisiert am 03.05.2019",
-            "url": "https://daten.berlin.de/datensaetze/wochen-und-tr%C3%B6delm%C3%A4rkte-0"
-        }
-    },
```
Comment (johnjohndoe) [inline cities/berlin.json]: The `metadata` block cannot be removed. It is still added manually, see this [`README.md`](https://github.com/wo-ist-markt/wo-ist-markt.github.io/blob/master/preprocessing/README.md).

File: preprocessing/berlin/compile-berlin-geojson.js [hunk 2]
```diff
@@ -11,6 +11,17 @@ var opening_hours = require('opening_hours');
...
 var OUTPUT_FILE = "./cities/berlin.json";
 
+// Sort input by title
+fs.readFile(INPUT_FILE, 'utf8', function (error, data) {
+	if (error) {
+		throw error;
+	}
+	var featureCollection = JSON.parse(data);
+	var outputData = featureCollection;
+  outputData.features = outputData.features.sort(function(a,b){return a.properties.title > b.properties.title ? 1 : -1;});
```
Comment (johnjohndoe) [inline preprocessing/berlin/compile-berlin-geojson.js]: Formatting: Please adjust the indentation and add some whitespace within the function.

File: preprocessing/berlin/compile-berlin-geojson.js [hunk 3]
```diff
@@ -11,6 +11,17 @@ var opening_hours = require('opening_hours');
 var INPUT_FILE = "./preprocessing/berlin/raw/markets-berlin.json";
 var OUTPUT_FILE = "./cities/berlin.json";
 
+// Sort input by title
```
Comment (johnjohndoe) [inline preprocessing/berlin/compile-berlin-geojson.js]: You could wrap the code block in a function which has a descriptive title which lets the comment unneeded.

**Comment by NWuensche** [comment]
Thanks for the notes, everything should be cleaned-up now.

**Pull request merged**